Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Build stage
+FROM python:3.9-slim AS build
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1
+
+# Set work directory
+WORKDIR /app
+
+# Copy requirements and install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy project files
+COPY . .
+
+# Run stage
+FROM python:3.9-slim AS run
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1
+
+# Set work directory
+WORKDIR /app
+
+# Copy dependencies from build stage
+COPY --from=build /usr/local/lib/python3.9/site-packages /usr/local/lib/python3.9/site-packages

```

Comment:
When copying dependencies from the build stage, ensure that the path '/usr/local/lib/python3.9/site-packages' is correct and matches where the dependencies are installed in the build stage. Otherwise, this could lead to runtime errors if the dependencies are not found.